percona/controllers/version/vs.go

```diff
@@ -0,0 +1,135 @@
+package version
+
+import (
+	"net/http"
+	"net/url"
+	"time"
+
+	api "github.com/percona/percona-postgresql-operator/pkg/apis/crunchydata.com/v1"
+	"github.com/percona/percona-postgresql-operator/versionserviceclient"
+	"github.com/percona/percona-postgresql-operator/versionserviceclient/models"
+	"github.com/percona/percona-postgresql-operator/versionserviceclient/version_service"
+	"github.com/pkg/errors"
+)
+
+const productName = "postgresql-operator"
+
+func (vs VersionServiceClient) GetExactVersion(cr *api.PerconaPGCluster, endpoint string, vm versionMeta) (DepVersion, error) {
+	if len(endpoint) == 0 {
+		endpoint = "https://check.percona.com"
+	}
+	requestURL, err := url.Parse(endpoint)
+	if err != nil {
+		return DepVersion{}, errors.Wrap(err, "parse endpoint")
+	}
+
+	srvCl := versionserviceclient.NewHTTPClientWithConfig(nil, &versionserviceclient.TransportConfig{
+		Host:     requestURL.Host,
+		BasePath: requestURL.Path,
+		Schemes:  []string{requestURL.Scheme},
+	})
+
+	applyParams := &version_service.VersionServiceApplyParams{
```

Comment:
We need to be more informative e.g. print in operator log that smart update was started/finished, also print what versions we get from check.percona.com.